RP Cloud Overview and Security doc (#1504)
redpanda-data/documentation: docs/deploy/deployment-option/cloud/security/cloud-security-architecture/cloud-encryption.mdx

---
title: Encryption
---

<head>
    <meta name="title" content="Redpanda Cloud Encryption  | Redpanda Docs"/>
    <meta name="description" content="Learn how Redpanda Cloud provides data encryption in transit and at rest."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/security/cloud-security-architecture/cloud-encryption/" />
</head>

Redpanda Cloud provides data at rest and data in transit encryption.

## Data at rest encryption

For data on disk, Redpanda Cloud relies on the cloud provider's default volume
encryption. The default encryption uses AES-256 block cipher and encryption keys
either per disk or data chunk, depending on the cloud provider.

> For details about how default data at rest encryption works on AWS or GCP, see

Reviewer: ```suggestion
For details about how default data at rest encryption works, see
```

---
title: Encryption
---

<head>
    <meta name="title" content="Redpanda Cloud Encryption  | Redpanda Docs"/>
    <meta name="description" content="Learn about Redpanda Cloud encryption at rest and encryption in transit."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/security/cloud-security-architecture/cloud-encryption/" />
</head>

> Redpanda Cloud provides data at rest and in transit encryption.

Reviewer 2: Redpanda Clodu provides data encryption in transit and at rest.

> Redpanda Cloud provides data at rest and in transit encryption.

Reviewer: ```suggestion
Redpanda Cloud provides data at rest and data in transit encryption.
```

> ## Data at rest encryption

Author: Not sure I agree with this one--in most companies where I've documented security, it has always been referred to as data at rest...see https://docs.cloudera.com/cdp-private-cloud-base/7.1.4/security-encrypting-data-at-rest/topics/cm-security-encryption-planning.html and https://docs.confluent.io/cloud/current/clusters/byok/index.html (first line reads  "You can encrypt data at rest in Dedicated Kafka clusters with self-managed keys to ensure only the appropriate entity or user can decrypt it..."

For data on disk, Redpanda Cloud relies on the cloud provider's default volume

> encryption. The default encryption uses AES=256 block cipher and encryption keys

Reviewer 3: Just a typo, am sure.

either per disk or data chunk, depending on the cloud provider.

> For details about how default data at rest encryption works on AWS or GCP, refer
> to the [AWS SSD instance store volume](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ssd-instance-store.html)
> or [GCP default encryption at rest](https://cloud.google.com/docs/security/encryption/default-encryption)
> product documentation.

Reviewer: ```suggestion
For details about how default data at rest encryption works on AWS or GCP, see [AWS SSD instance store volume](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ssd-instance-store.html)
or [GCP default encryption at rest](https://cloud.google.com/docs/security/encryption/default-encryption).
```

For Tiered Storage data, every Redpanda Cloud cluster uses a unique

> and periodically rotated managed master key (SSE-S3). The block cipher used is AES-256.

Reviewer: ```suggestion
and periodically rotated managed master key (SSE-S3). The block cipher uses AES-256.
```

> ## Data in transit encryption

Reviewer 2: In transit data encryption

Reviewer 2: that's how I have most commonly read them in the wild.

Author: See my response above: https://github.com/redpanda-data/documentation/pull/1504/files#r1237388292

All network traffic transporting customer data is encrypted in transit using
asymmetric encryption with TLS 1.2. The network connection to the control plane
is also TLS 1.2 encrypted.

Data plane TLS certificates are generated and signed by
[Let’s Encrypt](https://letsencrypt.org/). Redpanda Cloud implements mitigations

> to prevent ill-intentioned actors from enumerating cluster endpoints through

Reviewer: ```suggestion
to prevent bad actors from enumerating cluster endpoints through
```

the public certificate transparency log.

> Following are the protocols and cipher suites supported and accepted by Redpanda

Reviewer: ```suggestion
The following protocols and cipher suites are supported and accepted by Redpanda
```

Accepted  TLSv1.1  128 bits  ECDHE-RSA-AES128-SHA          Curve 25519 DHE 253
Accepted  TLSv1.1  128 bits  DHE-RSA-AES128-SHA            DHE 2048 bits
Accepted  TLSv1.1  256 bits  AES256-SHA
Accepted  TLSv1.1  128 bits  AES128-SHA
  # highlight-next-line
Preferred TLSv1.0  256 bits  ECDHE-RSA-AES256-SHA          Curve 25519 DHE 253
Accepted  TLSv1.0  256 bits  DHE-RSA-AES256-SHA            DHE 2048 bits
Accepted  TLSv1.0  128 bits  ECDHE-RSA-AES128-SHA          Curve 25519 DHE 253
Accepted  TLSv1.0  128 bits  DHE-RSA-AES128-SHA            DHE 2048 bits
Accepted  TLSv1.0  256 bits  AES256-SHA
Accepted  TLSv1.0  128 bits  AES128-SHA
Server Key Exchange Group(s):
TLSv1.3  128 bits  secp256r1 (NIST P-256)
TLSv1.3  192 bits  secp384r1 (NIST P-384)
TLSv1.3  260 bits  secp521r1 (NIST P-521)
TLSv1.3  128 bits  x25519
TLSv1.3  224 bits  x448
TLSv1.3  112 bits  ffdhe2048
TLSv1.3  128 bits  ffdhe3072
TLSv1.3  150 bits  ffdhe4096
TLSv1.3  175 bits  ffdhe6144
TLSv1.3  192 bits  ffdhe8192
TLSv1.2  128 bits  secp256r1 (NIST P-256)
TLSv1.2  192 bits  secp384r1 (NIST P-384)
TLSv1.2  260 bits  secp521r1 (NIST P-521)
TLSv1.2  128 bits  x25519

> TLSv1.2  224 bits  x448

Reviewer 3: Might be a good discussion to take up and drive outside of this PR, Chase.

> TLSv1.2  224 bits  x448

Author: @bpraseed or @towfiqa Can you comment/advise? thx
